### PR TITLE
add logout function

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
This PR implements the Logout functionality and adds a Logout confirmation modal. The functionality includes:

### **useLogout Hook:**

Handles the logout process by:
Removing the authentication token from local storage.
Redirecting the user to the Sign In page (/auth/signInPage).
Resetting or invalidating the RTK Query cache to clear the user's session data.

### **LogoutModal Component:**

A confirmation modal that is triggered when the user clicks on the "Logout" button.
The modal prompts the user to confirm whether they want to log out of the application.
If the user confirms, the useLogout hook is triggered, logging the user out and redirecting them to the Sign In page.
If the user cancels, they remain on the current page.

### **Changes:**

Added the useLogout hook to handle the full logout flow.
Created the LogoutModal component to provide a smooth UX for logging out.
Integrated the modal into the sidebar’s logout button.

### CODE REVIEW

**Comment by EvgeniyaYakunina** [comment]
run build production

File: src/features/auth/api/authApi.ts [hunk 1]
```diff
@@ -39,6 +39,14 @@ export const authApi = baseApi.injectEndpoints({
           url: API_URLS.AUTH.NEW_PASSWORD,
         }),
       }),
+      logout: builder.mutation<void, void>({
```
Comment (Lelik7777) [inline src/features/auth/api/authApi.ts:42]: add js doc comments

File: src/features/auth/ui/logout/logoutModal/logoutModal.module.scss [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    padding: 12px;
+  }
+}
+
+.content {
+  width: 100%;
+  max-width: 380px;
+
+  @include mobile {
+    max-width: 330px;
```
Comment (Lelik7777) [inline src/features/auth/ui/logout/logoutModal/logoutModal.module.scss:30]: --mobile-content-width: 330px;
use

Comment (Lelik7777) [inline src/features/auth/ui/logout/logoutModal/logoutModal.module.scss:30]: --mobile-content-width: 330px;
use

File: src/features/auth/ui/logout/logoutModal/logoutModal.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  ModalHeader,
+  ModalTitle,
+  Typography,
+} from '@photo-fiesta/ui-lib'
+
+import styles from './logoutModal.module.scss'
+export type LogoutModalProps = {
+  closeModal: () => void
+  confirmLogout: () => void
+  open: boolean
```
Comment (Lelik7777) [inline src/features/auth/ui/logout/logoutModal/logoutModal.tsx:17]: isOpen

File: src/features/auth/ui/logout/useLogout.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import { baseApi } from '@/app/api'
+import { Storage } from '@/shared/utils/storage'
+import { useRouter } from 'next/router'
+
+export const useLogout = () => {
+  const router = useRouter()
+  const dispatch = useDispatch()
+
+  return () => {
```
Comment (Lelik7777) [inline src/features/auth/ui/logout/useLogout.tsx:11]: add js doc comments

File: src/features/auth/ui/logout/useLogout.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { useDispatch } from 'react-redux'
```
Comment (Lelik7777) [inline src/features/auth/ui/logout/useLogout.tsx:1]: add index.ts in logout 
add index.ts in logoutModal
add export from logout in index.ts in ui

File: src/shared/ui/sidebar/sidebar.tsx [hunk 6]
```diff
@@ -14,6 +18,18 @@ import Link from 'next/link'
 import styles from './sidebar.module.scss'
 
 export const Sidebar = () => {
+  const [isModalOpen, setModalOpen] = useState(false)
```
Comment (Lelik7777) [inline src/shared/ui/sidebar/sidebar.tsx:21]: setIsModalOpen

File: src/features/auth/ui/logout/logoutModal/logoutModal.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+export const LogoutModal = ({ closeModal, confirmLogout, open }: LogoutModalProps) => {
```
Comment (Lelik7777) [inline src/features/auth/ui/logout/logoutModal/logoutModal.tsx:20]: add js doc comments with example

File: src/shared/ui/sidebar/sidebar.tsx [hunk 8]
```diff
@@ -14,6 +18,18 @@ import Link from 'next/link'
...
+
+  const confirmLogout = () => {
+    logout()
+    setModalOpen(false)
+  }
+
+  const closeModal = () => {
+    setModalOpen(false)
+  }
+
```
Comment (Lelik7777) [inline src/shared/ui/sidebar/sidebar.tsx:32]: можно вынести в хук useSideBar

File: src/shared/ui/sidebar/sidebar.tsx [hunk 9]
```diff
@@ -14,6 +18,18 @@ import Link from 'next/link'
 import styles from './sidebar.module.scss'
 
 export const Sidebar = () => {
```
Comment (Lelik7777) [inline src/shared/ui/sidebar/sidebar.tsx:20]: add js doc comment with example

File: src/shared/ui/sidebar/sidebar.tsx [hunk 10]
```diff
@@ -14,6 +18,18 @@ import Link from 'next/link'
...
+  const confirmLogout = () => {
+    logout()
+    setModalOpen(false)
+  }
+
+  const closeModal = () => {
+    setModalOpen(false)
+  }
+
   return (
```
Comment (Lelik7777) [inline src/shared/ui/sidebar/sidebar.tsx:33]: use classNames

File: src/shared/ui/sidebar/sidebar.tsx [hunk 11]
```diff
@@ -64,12 +80,15 @@ export const Sidebar = () => {
           </Link>
         </div>
```
Comment (Lelik7777) [inline src/shared/ui/sidebar/sidebar.tsx:81]: add TODO: add routes constants

**Pull request merged**